src/lib.rs

```diff
@@ -2178,6 +2184,25 @@ impl Build {
                         }
 
                         cmd.push_cc_arg(format!("--target={}", target).into());
+                    } else if target.contains("wasm32") && target.contains("linux") {
+                        // wasm32-linux currently uses the LLVM-18 wasi threads target
+                        cmd.push_cc_arg("--target=wasm32-wasi-threads".into());
+                        for x in &[
+                            "atomics",
+                            "bulk-memory",
+                            "mutable-globals",
+                            "sign-ext",
+                            "exception-handling",
+                        ] {
+                            cmd.push_cc_arg(format!("-m{}", x).into());
+                        }
+                        for x in &["wasm-exceptions", "declspec"] {
+                            cmd.push_cc_arg(format!("-f{}", x).into());
+                        }
+                        let musl_sysroot = self.wasm_musl_sysroot().unwrap();
+                        cmd.push_cc_arg(
+                            format!("--sysroot={}", Path::new(&musl_sysroot).display()).into(),
+                        );
```

Comment:
cc @madsmtm do you think we can use gen-target-info to auto generate this information?

Reply:
Hmm, the `--target` at least definitely looks odd?